loclist.kak

```diff
@@ -0,0 +1,147 @@
+        loclist-jump
+    }
+    try %{
+        evaluate-commands -client %opt{toolsclient} %{
+            buffer %opt{loclist_buffer}
+            execute-keys gg %opt{grep_current_line}g
+        }
+    }
+}
+
+define-command loclist-previous-location -docstring %{
+    loclist-previous-location
+    Jump to previous location listed in the current location list buffer, usually one of
+    *diagnostics* *goto* *grep* *implementations* *lint-output* *make* *references* *symbols*
+
+    %opt{loclist_buffer} determines the buffer current location list buffer.
+    %opt{loclist_location_format} determines matching locations.
+} -buffer-completion %{
+    evaluate-commands -try-client %opt{jumpclient} -save-regs / %{
+        buffer %opt{loclist_buffer}
+        set-register / %opt{loclist_location_format}
+        execute-keys ge %opt{grep_current_line}g<a-h> <a-/><ret>
+        loclist-jump
+    }
+    try %{
+        evaluate-commands -client %opt{toolsclient} %{
+            buffer %opt{loclist_buffer}
+            execute-keys gg %opt{grep_current_line}g
+        }
+    }
+}
+
+define-command -override loclist-stack-push -docstring "record location list buffer" %{
+    evaluate-commands %sh{
+        eval set -- $kak_quoted_opt_loclist_stack
+        if printf '%s\n' "$@" | grep -Fxq -- "$kak_bufname"; then {
+            exit
+        } fi
+        newbuf=$kak_bufname-$#
+        echo "try %{ delete-buffer! $newbuf }"
+        echo "rename-buffer $newbuf"
+        echo "set-option -add global loclist_stack %val{bufname}"
+    }
+    set-option global loclist_buffer %val{bufname}
+}
+
+define-command -override loclist-stack-pop -docstring "restore location list buffer" %{
+    evaluate-commands %sh{
+        eval set -- $kak_quoted_opt_loclist_stack
+        if [ $# -eq 0 ]; then {
+            echo fail "loclist-stack-pop: no location list buffer to pop"
+            exit
+        } fi
+        printf 'set-option global loclist_stack'
+        top=
+        while [ $# -ge 2 ]; do {
+            top=$1
+            printf ' %s' "$1"
+            shift
+        } done
+        echo
+        echo "delete-buffer $1"
+        echo "set-option global loclist_buffer '$top'"
+    }
+    try %{
+        evaluate-commands -try-client %opt{jumpclient} %{
+            buffer %opt{loclist_buffer}
+            grep-jump
+        }
+    }
+}
+define-command -override loclist-stack-clear -docstring "clear location list buffers" %{
+    evaluate-commands %sh{
+        eval set --  $kak_quoted_opt_loclist_stack
+        printf 'try %%{ delete-buffer %s }\n' "$@"
+    }
+    set-option global loclist_stack
+}
+
+# TODO: What exactly does this disable, and do we need to?
+set-option global disabled_hooks "%opt{disabled_hooks}|grep-jump"
+declare-option regex loclist_buffer_regex \*(?:callees|callers|diagnostics|goto|find|grep|implementations|lint-output|references)\*(?:-\d+)?
+
+hook -group loclist global WinDisplay %opt{loclist_buffer_regex} %{
```

Comment:
I recently saw a [different approach](https://github.com/shachaf/kak/blob/98078d51c61bfab3daa79b7a28d28d030ad4181e/kakrc#L209-L211) using the `BufOpenFifo` hook.
Maybe we should do that.
Then we would only change the option when we either
create a new locations buffer, or
pop the location stack.

The upside is that it would be less magic.
I need to try it; I think I rarely ever rely on the extra `WinDisplay` magic.